google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/SampleRowKeysCallableTest.java

```diff
@@ -0,0 +1,126 @@
+import com.google.api.gax.rpc.ApiCallContext;
+import com.google.api.gax.rpc.NotFoundException;
+import com.google.api.gax.rpc.UnaryCallable;
+import com.google.bigtable.admin.v2.InstanceName;
+import com.google.bigtable.v2.SampleRowKeysRequest;
+import com.google.bigtable.v2.SampleRowKeysResponse;
+import com.google.cloud.bigtable.data.v2.internal.RequestContext;
+import com.google.cloud.bigtable.data.v2.wrappers.KeyOffset;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.ByteString;
+import io.grpc.Status.Code;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class SampleRowKeysCallableTest {
+  private final RequestContext requestContext =
+      RequestContext.create(InstanceName.of("my-project", "my-instance"), "my-app-profile");
+  private FakeCallable inner;
+  private SampleRowKeysCallable callable;
+
+  @Before
+  public void setUp() {
+    inner = new FakeCallable();
+    callable = new SampleRowKeysCallable(inner, requestContext);
+  }
+
+  @Test
+  public void requestIsCorrect() {
+    callable.futureCall("my-table");
+
+    assertThat(inner.request)
+        .isEqualTo(
+            SampleRowKeysRequest.newBuilder()
+                .setTableName(requestContext.getInstanceName() + "/tables/my-table")
+                .setAppProfileId(requestContext.getAppProfileId())
+                .build());
+  }
+
+  @Test
+  public void responseCorrectlyTransformed()
+      throws ExecutionException, InterruptedException, TimeoutException {
+    ApiFuture<List<KeyOffset>> result = callable.futureCall("my-table");
+
+    inner.response.set(
+        ImmutableList.of(
+            SampleRowKeysResponse.newBuilder()
+                .setRowKey(ByteString.copyFromUtf8("key1"))
+                .setOffsetBytes(100)
+                .build(),
+            SampleRowKeysResponse.newBuilder()
+                .setRowKey(ByteString.copyFromUtf8(""))
+                .setOffsetBytes(1000)
+                .build()));
+
+    assertThat(result.get(1, TimeUnit.SECONDS))
+        .isEqualTo(
+            ImmutableList.of(
+                KeyOffset.create(ByteString.copyFromUtf8("key1"), 100),
+                KeyOffset.create(ByteString.EMPTY, 1000)));
+  }
+
+  @Test
+  public void errorIsPropagated() {
+    ApiFuture<List<KeyOffset>> result = callable.futureCall("my-table");
+
+    Throwable expectedError =
+        new NotFoundException("fake error", null, GrpcStatusCode.of(Code.NOT_FOUND), false);
+    inner.response.setException(expectedError);
+
+    Throwable actualError = null;
+    try {
+      result.get(1, TimeUnit.SECONDS);
+    } catch (ExecutionException e) {
+      actualError = e.getCause();
+    } catch (Throwable t) {
+      actualError = t;
+    }
```

Comment:
I don't understand here, do we expect actualError from e.getCause() or t?

Reply:
we want the cause of the execution exception. I removed the rest of the noise